go/arrow/example_test.go

```diff
@@ -17,11 +17,14 @@
 package arrow_test
 
 import (
+	"context"
 	"fmt"
 	"log"
 
 	"github.com/apache/arrow/go/v13/arrow"
 	"github.com/apache/arrow/go/v13/arrow/array"
+	"github.com/apache/arrow/go/v13/arrow/compute"
+	"github.com/apache/arrow/go/v13/arrow/compute/exec"
```

Comment:
These will only work with go1.18 (because of build constraints). I think it makes most sense for this example to exist at least inside of the compute directory as a test location, and make sure to include the build constraint for `//go:build go1.18`